### PR TITLE
added config/maps cmd line args,

### PR DESCRIPTION
**Pull request by kvsikand**
This allows for launching enml from a different working directory.

Putting this up as a pull request because I had a slight lack of confidence in the way I handled creating the `NonMarkovLocalization` instance. Specifically, I made it a pointer to allow for initializing it only after the arguments get parsed, since the maps_dir_ is taken in by the constructor.

I'm not sure if this is a "best practice", so let me know if you have an alternative suggestion (the other option that came to my mind was to initialize without a map directory, and add a setter on the `NonMarkovLocalization` for setting it after the configs are loaded. This would require removing the `const` modifier from  `maps_dir_` in `non_markov_localization.h`)

### CODE REVIEW

**Comment by joydeep-b** [comment]
This PR seems to be missing some of the changes @maxsvetlik had, including handling the directory for maps as well. Should we use the other PR instead? Or is the other one missing something from here? Please consolidate, @maxsvetlik @kvsikand

**Comment by joydeep-b** [comment]
There are also changes to the config file, are all of them intentional? Please review carefully.

**Comment by kvsikand** [comment]
This PR does handle the maps stuff, its the `maps_dir` cmd-line flag. The other PR doesn't work in the case where enml is a metapackage, as it is in the ut-jackal repo.
The config file changes were the result of a merge, I'll roll them back.

File: config/enml.lua [hunk 1]
```diff
@@ -90,25 +90,25 @@ if RobotConfig.name=="ut-jackal" or RobotConfig.name=="smads-ut-jackal" then
...
 
-  -- -- 2020-06-25-17-34-30.bag
+  -- 2020-06-25-17-34-30.bag
   -- enml.map_name = "UT_Campus";
   -- enml.starting_loc_x = 4;
   -- enml.starting_loc_y = -29;
   -- enml.starting_angle = deg2rad(-122);
 
-  -- -- 2020-06-03-18-51-39.bag 160 seconds in
+  -- 2020-06-25-17-34-30.bag 160 seconds in
```
[the remark this answers is not in the document]
Comment (kvsikand) [reply config/enml.lua:99]: No, and in fact I don't think it's actually changing it. I have compared my file to what's on master currently, and it's identical throughout this section. Something strange is going on...looks like the thing this github gui is comparing against is not actually what's on master at the moment

**Pull request merged**